backend/modules/simulation/services/grading_service.py

```
@@ -0,0 +1,213 @@
+            # Fallback if grading agent not available
+            return {
+                "overall_score": 0,
+                "overall_feedback": "Grading agent not available. Please try again later.",
+                "scenes": [],
+                "rubric_total_points": 100
+            }
+        
+        # Get scenario and scenes
+        scenario = self.repository.get_scenario_by_id(user_progress.scenario_id)
+        if not scenario:
+            raise NotFoundError("Scenario not found")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)